financial-services-accelerator/consent-mgt-api/org.wso2.financial.services.accelerator.consent.mgt.api.service/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/api/service/ConsentCoreService.java

```diff
@@ -0,0 +1,306 @@
+    boolean storeConsentAttributes(String consentId, Map<String, Object> consentAttributes)
+            throws ConsentMgtException;
+
+    /**
+     * This method is used to get consent attributes for a provided attribute keys list related to a particular consent.
+     *
+     * @param consentId            consent ID
+     * @param consentAttributeKeys consent attribute keys list
+     * @return a consent attributes resource
+     * @throws ConsentMgtException thrown if an error occurs in the process
+     */
+    ConsentAttributes getConsentAttributes(String consentId, ArrayList<String> consentAttributeKeys)
+            throws ConsentMgtException;
+
+    /**
+     * This method is used to get consent attributes related to a particular consent.
+     *
+     * @param consentId consent ID
+     * @return a consent attributes resource
+     * @throws ConsentMgtException thrown if an error occurs in the process
+     */
+    ConsentAttributes getConsentAttributes(String consentId) throws
+            ConsentMgtException;
+
+    /**
+     * This method is used to delete the provided consent attributes for a particular consent.
+     *
+     * @param consentId         consent ID
+     * @param consentAttributes attributes to update
+     * @return updated consent attributes
+     * @throws ConsentMgtException thrown if an error occurs in the process
+     */
+    ConsentAttributes updateConsentAttributes(String consentId, Map<String, Object> consentAttributes)
+            throws ConsentMgtException;
+
+    /**
+     * This method is used to delete the provided consent attributes for a particular consent.
+     *
+     * @param consentId         consent ID
+     * @param attributeKeysList attributes to delete
+     * @return true if deletion is successful
+     * @throws ConsentMgtException thrown if an error occurs in the process
+     */
+    boolean deleteConsentAttributes(String consentId, ArrayList<String> attributeKeysList)
+            throws ConsentMgtException;
+
+    /**
+     * This method is used to search audit records. Useful for auditing purposes. All the input parameters are
+     * optional. If all parameters are null, all the audit records will be returned.
+     *
+     * @param consentId     consent ID
+     * @param status        status of the audit records needed
+     * @param actionBy      user who performed the status change
+     * @param fromTime      from time
+     * @param toTime        to time
+     * @param statusAuditID ID of a specific audit record that need to be searched
+     * @return a list of consent status audit records
+     * @throws ConsentMgtException thrown if an error occurs
+     */
+    ArrayList<ConsentStatusAuditRecord> searchConsentStatusAuditRecords(String consentId, String status,
+                                                                        String actionBy, Long fromTime, Long toTime,
+                                                                        String statusAuditID)
```

Comment:
Check other places as well
```suggestion
                                                                        String statusAuditId)
```

Reply:
resolved in https://github.com/wso2/financial-services-accelerator/pull/600/commits/a6406d8d973f00c253e84310b1d02ca6d62cedd7